### PR TITLE
castor dqm update

### PR DESCRIPTION
**Pull request by popovvp**
- rm CastorBaseMonitor module
- removed some TH2D histograms :
    instead of 2xTH2D add TProfile

- add Energy (digi) vs BX (latency)
- re-implement DB conditions for masking

### CODE REVIEW

**Comment by andrius-k** [comment]
Hi,

Are you sure that code checks in this case didn't pass because of the bot error? Could you please review the code one more time?

**Comment by popovvp** [comment]
Hi,
  not sure what kind of error you mean.
  I had no problem in CMSSW_10_2_0_pre5/

**Comment by popovvp** [comment]
Hi,
  checked again in version 10_3_0_pre1 (run320317 RAW): works without
problem, but trigger & digi Collections are empty
  (in version 10_2_0_pre5 all Collections filled)

**Comment by andrius-k** [comment]
Hi,

Yes the code seems to be fine functionally, but code checks still fail because of the styling issues. In particular it seems that you removed override statements from the methods which are actually overriding base class methods. Please see [this](https://github.com/cms-sw/cmssw/pull/24142#issuecomment-409190069) bot comment. Override statements usually have no impact on functionality but they are good tool to avoid errors in the future, and they are required by CMSSW policy.

**Comment by davidlange6** [comment]
please test

File: DQM/CastorMonitor/src/CastorDigiMonitor.cc [hunk 1]
```diff
@@ -31,12 +53,31 @@ void CastorDigiMonitor::bookHistograms(DQMStore::IBooker& ibooker,
...
   char sTileIndex[50];
-  sprintf(sTileIndex,"Tile(=moduleZ*16+sector#phi)");
+  sprintf(sTileIndex,"Cell(=moduleZ*16+sector#phi)");
 
   ievt_=0;
 
+  ibooker.setCurrentFolder(subsystemname_);
+  hBX = ibooker.bookProfile("average E(digi) in BX",
+    "Castor average E (digi);Event.BX;fC", 3601,-0.5,3600.5, 0.,1.e10,"");
+  hBX->getTProfile()->SetOption("hist");
```
Comment (andrius-k) [inline DQM/CastorMonitor/src/CastorDigiMonitor.cc]: Please avoid using root objects directly. The option that you are setting will have no effect on the final plot in DQM GUI. In order to set a draw option you either have to use Customise menu in DQM GUI or add render plugin to DQM GUI.

File: DQM/CastorMonitor/src/CastorDigiMonitor.cc [hunk 2]
```diff
@@ -81,38 +122,53 @@ void CastorDigiMonitor::bookHistograms(DQMStore::IBooker& ibooker,
...
+  sprintf(s,"CASTOR_Tower_EMvsEhad(fC)");
+    h2towEMvsHAD = ibooker.book2D(s,s, NEtow, EhadTow, NEtow, EMTow);
+    h2towEMvsHAD->getTH2F()->GetXaxis()->SetTitle("Ehad [fC]");
+    h2towEMvsHAD->getTH2F()->GetYaxis()->SetTitle("EM [fC]");
+    h2towEMvsHAD->getTH2F()->SetOption("colz");
+
+ sprintf(s,"CASTOR_TowerTotalEnergy(fC)");
+   htowE = ibooker.book1D(s,s,NEtow+1,ETower);
+   htowE->getTH1F()->GetXaxis()->SetTitle("fC");
+
```
Comment (andrius-k) [inline DQM/CastorMonitor/src/CastorDigiMonitor.cc]: Please take a look at the indentation over here. Lines bellow `sprintf`s should be on the same level as `sprintf`s themselves.

File: DQM/CastorMonitor/src/CastorMonitorModule.cc [hunk 3]
```diff
@@ -65,30 +71,52 @@ CastorMonitorModule::~CastorMonitorModule() {
...
+	chInd,HcalCastorDetId(*ch).module(),HcalCastorDetId(*ch).sector());
+    for (short unsigned int iCapId = 0; iCapId < 4; iCapId++){
+      printf("%f ",sqrt(pedestal->getWidth(iCapId)));
+      sigma_avr += sqrt(pedestal->getWidth(iCapId));
+    }
+    printf("= %f\n",sigma_avr);
+fPedSigma[HcalCastorDetId(*ch).module()-1][HcalCastorDetId(*ch).sector()-1]=sigma_avr/4;
+  } // end for (std::vector<DetId>::iterator ch=channels.begin();...
+*/
 }
```
Comment (andrius-k) [inline DQM/CastorMonitor/src/CastorMonitorModule.cc]: Are you sure that this commented out code is necessary here?

File: DQM/CastorMonitor/src/CastorRecHitMonitor.cc [hunk 4]
```diff
@@ -192,24 +196,26 @@ void CastorRecHitMonitor::processEvent(const CastorRecHitCollection& castorHits)
...
-
+/*
   if(ievt_ %100 == 0) 
     for(int mod=1; mod<=14; mod++) 
       for(int sec=1; sec<=16;sec++) {
 	double a= h2RHmap->getTH2F()->GetBinContent(mod,sec);
 	h2RHoccmap->getTH2F()->SetBinContent(mod,sec,a/double(ievt_));
       }
-
+*/
```
Comment (andrius-k) [inline DQM/CastorMonitor/src/CastorRecHitMonitor.cc]: Are you sure that this block of commented out code is necessary here?

**Comment by andrius-k** [comment]
Hi,

I made some comments in the changes and also I think it would be a really good idea to apply automatic code formatting for the code because it's readability would increase vastly. Could you please apply `clang-format --style=google -i <file-name>` for every file that you changed?

File: DQM/CastorMonitor/src/CastorRecHitMonitor.cc (hunk 4 above)
Comment (popovvp) [reply DQM/CastorMonitor/src/CastorRecHitMonitor.cc]: cleaned commented out lines:
CastorRecHitMonitor.cc
CastorDigiMonitor.cc, CastorDigiMonitor.h
CastorMonitorModule.cc

**Comment by davidlange6** [comment]
please test

**Comment by andrius-k** [comment]
Hi,

Did you apply `clang-format` for the files that you changed in the latest two commits?

**Comment by andrius-k** [comment]
Hi,

Could you please apply `clang-format` to the files that you changed? Now the formatting could be improved a bit and that would significantly increase the readability of the code.

**Comment by popovvp** [comment]
Hi,

  where this instruction have to be inserted to
and how the instruction looks.
  Could you provide with example.

  thanks,

**Comment by andrius-k** [comment]
Hi,

First you would have to navigate to the source directory like so: `cd CMSSW_10_2_0_pre5/src/`
Then you would just have to run this command: 
`clang-format --style=google -i DQM/CastorMonitor/src/CastorMonitorModule.cc DQM/CastorMonitor/src/CastorRecHitMonitor.cc DQM/CastorMonitor/src/CastorMonitorModule.cc DQM/CastorMonitor/src/CastorRecHitMonitor.cc`
And in the end please commit and push the changes.

**Comment by andrius-k** [comment]
please test

**Comment by andrius-k** [comment]
+1

**Comment by andrius-k** [comment]
Thank you for applying the the styling rules. Now code looks much nicer!

**Comment by kpedro88** [comment]
+1

**Comment by fabiocos** [comment]
@slava77 thanks for catching the problem

**Pull request merged**